R/plotLoadings.R

```diff
@@ -0,0 +1,362 @@
+#'
+#' Inspired by the \code{\link[diffTop:plotASVcircular]{plotASVcircular}} method using phyloseq 
+#' and has been converted to use TreeSummarizedExperiment objects.
+#'
+#' @param x a
+#'   \code{\link[TreeSummarizedExperiment:TreeSummarizedExperiment-class]{TreeSummarizedExperiment}}
+#'   x.
+#' 
+#' @param dimred \code{Character scalar}. Dimension reduction from reducedDimNames to be plotted. 
+#'   (default: \code{dimred = "PCA"})
+#'  
+#' @param layout One way to plot feature loadings of \code{c("heatmap", "barplot", "screeplot", "tree")} 
+#'   (default: \code{layout = "heatmap"})
+#' 
+#' @param n A numeric specifying the number of features to be plotted.
+#'   (default: \code{n = 10})
+#'   
+#' @param ncomponents A numeric specifying the number of components.
+#'   (default: \code{ncomponents = 5})
+#' 
+#' @param tree.name a single \code{character} value specifying a rowTree from
+#'   TreeSummarizedExperiment. (By default: \code{tree.name = "phylo"})
+#'     
+#' 
+#' @details
+#' It is impossible to plot tree if only the matrix is given. Number of features must be reduced
+#' before calling function or it will not be understandable. 
+#' 
+#' @return 
+#' A \code{ggplot2} object. A circular plot annotated with TreeSummarizedExperiment object.
+#'
+#' @name plotLoadings
+#' @export
+#' 
+#' @author
+#' Ely Seraidarian
+#' Contact: \url{microbiome.github.io}
+#'
+#' @examples
+#' 
+#' # Plotting feature loadings with tree
+#' library(mia)
+#' library(ggtree)
+#' library(scater)
+#' data("GlobalPatterns", package = "mia")
+#' tse <- GlobalPatterns
+#' tse <- transformAssay(tse, method = "clr", pseudocount = 1)
+#' tse <- agglomerateByPrevalence(tse, rank="Phylum", update.tree = TRUE)
+#' tse <- runPCA(tse, ncomponents = 5, assay.type = "clr")
+#' plotLoadings(tse, layout = "tree")
+#' 
+#' # Plotting without tree as a heatmap
+#' loadings_matrix <- attr(reducedDim(tse, "PCA"), "rotation")
+#' plotLoadings(loadings_matrix, layout = "heatmap")
+#' 
+#' # Plotting without tree as a barplot
+#' plotLoadings(loadings_matrix, layout = "barplot")
+#' 
+#' # Plotting without tree as a screeplot
+#' plotLoadings(loadings_matrix, layout = "screeplot")
+#' 
+#' # Plotting more features
+#' plotLoadings(loadings_matrix, layout = "heatmap", n = 12)
+#'
+#' # Plotting with less components
+#' tse <- runPCA(tse, ncomponents = 4, assay.type = "clr")
+#' loadings_matrix <- attr(reducedDim(tse, "PCA"), "rotation")
+#' plotLoadings(loadings_matrix, layout = "heatmap", ncomponents = 4 )
+#' 
+#' # Plotting if loadings matrix name has been changed
+#' tse <- runPCA(tse, name = "myPCAmatrix", ncomponents = 5, assay.type = "clr")
+#' plotLoadings(tse, dimred= "myPCAmatrix", layout = "heatmap")
+NULL
+
+#' @rdname plotLoadings
+setGeneric("plotLoadings", signature = c("x"),
+    function(x, ...) 
+        standardGeneric("plotLoadings"))
+
+
+#' @rdname plotLoadings
+#' @export 
+setMethod("plotLoadings", signature = c(x = "TreeSummarizedExperiment"),
```

Comment:
We could enable the function for SingleCellExperiment objects also but then missing rowTree slot should be taken into acount (SCE objects do not have trees)